### PR TITLE
chore(rxForm): decompose monolithic source code

### PR DESCRIPTION
**Pull request by CITguy**
[FRMW-354](https://jira.rax.io/browse/FRMW-354)

### LGTMS
- [x] Dev LGTM
- [x] Dev LGTM

### CODE REVIEW

File: src/rxForm/rxForm.js [hunk 1]
```diff
@@ -1,904 +0,0 @@
...
- * ...
- * <form rx-form><!-- you can use a DIV, if desired -->
- *   <rx-form-section>
- *     ...
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxForm', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxForm.js`

File: src/rxForm/rxForm.js [hunk 2]
```diff
@@ -1,904 +0,0 @@
...
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- *
- * @param {*=} stacked
- * If present, `rxField` children will stack vertically rather than
- * display horizontally.
- */
-.directive('rxFormSection', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFormSection.js`

File: src/rxForm/rxForm.js [hunk 3]
```diff
@@ -1,904 +0,0 @@
...
- *     <rx-field>
- *       <rx-field-name>...</rx-field-name>
- *       <rx-field-content>...</rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxField', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxField.js`

File: src/rxForm/rxForm.js [hunk 4]
```diff
@@ -1,904 +0,0 @@
...
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- *
- * @param {Boolean=} [ng-required=false]
- * Is this field required? This will add/remove the required symbol to the left of the name.
- */
-.directive('rxFieldName', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFieldName.js`

File: src/rxForm/rxForm.js [hunk 5]
```diff
@@ -1,904 +0,0 @@
...
- *       <rx-field-content>
- *          <rx-input>...</rx-input>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxFieldContent', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFieldContent.js`

File: src/rxForm/rxForm.js [hunk 6]
```diff
@@ -1,904 +0,0 @@
...
- *           <input type="number" />
- *         </rx-input>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxInput', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxInput.js`

File: src/rxForm/rxForm.js [hunk 7]
```diff
@@ -1,904 +0,0 @@
...
- *           <rx-suffix>Million</rx-suffix>
- *         </rx-input>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxPrefix', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxPrefix.js`

File: src/rxForm/rxForm.js [hunk 8]
```diff
@@ -1,904 +0,0 @@
...
- *           <rx-suffix>Million</rx-suffix>
- *         </rx-input>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxSuffix', function (rxNestedElement) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxSuffix.js`

File: src/rxForm/rxForm.js [hunk 9]
```diff
@@ -1,904 +0,0 @@
...
- *           Salary must be above $1,000,000
- *         </rx-inline-error>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxInlineError', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxInlineError.js`

File: src/rxForm/rxForm.js [hunk 10]
```diff
@@ -1,904 +0,0 @@
...
- *           Salary must be above $1,000,000
- *         </rx-inline-error>
- *       </rx-field-content>
- *     </rx-field>
- *   </rx-form-section>
- * </form>
- * ...
- * </pre>
- */
-.directive('rxHelpText', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxHelpText.js`

File: src/rxForm/rxForm.js [hunk 11]
```diff
@@ -1,904 +0,0 @@
...
- * This directive is used to wrap input fields and select boxes in the proper HTML.
- * It will attach the `<label>` to a transcluded input using the `id` attribute of the input.
- * If no `id` attribute exists, it will create one.
- *
- * @param {String} label - Text to use for <label>
- * @param {String} prefix - Text to include to the left of content
- * @param {String} suffix - Text to include to the right of content
- * @param {String} description - Text to place below input
- */
-.directive('rxFormItem', function ($document, rxDOMHelper) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFormItem.js`

File: src/rxForm/rxForm.js [hunk 12]
```diff
@@ -1,904 +0,0 @@
...
- * **DEPRECATED** - This directive has been marked as deprecated and *will be removed* in a future
- * release of the EncoreUI framework.  Please see current rxForm documentation for updated
- * functionality.
- *
- * This directive is used to wrap a set of input fields in the proper HTML
- *
- * @param {String} legend - Text to use for <legend>
- * @param {String} description - Text to place below input
- */
-.directive('rxFormFieldset', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFormFieldset.js`

File: src/rxForm/rxForm.js [hunk 13]
```diff
@@ -1,904 +0,0 @@
...
- * // Returns the selected option for the rxFormOptionTable with id tableId
- * // [{ tableId: 'tableId', fieldId: 'fieldId', rowId: 'rowId' }]
- * getSelectedOptionForTable(tableId)
-
- * // Returns the selected option for the rxFormOptionTable in the tabset with id tabsetId
- * // [{ tableId: 'tableId', fieldId: 'fieldId', rowId: 'rowId' }]
- * getSelectedOptionForTabSet(tabsetId)
- * </pre>
- */
-.factory('rxFormUtils', function ($document) {
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: moved to `src/rxForm/scripts/rxFormUtils.js`

File: src/rxForm/rxForm.spec.js [hunk 14]
```diff
@@ -1,417 +0,0 @@
...
-        //module('encore.ui.rxMisc');
-
-        inject(function ($rootScope, $compile) {
-            rootScope = $rootScope;
-            scope = $rootScope.$new();
-            compile = $compile;
-        });
-    });
-
-    describe('(deprecated directive) rxFormItem', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxFormItem.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 15]
```diff
@@ -1,417 +0,0 @@
...
-
-    describe('hierarchy validation', function () {
-        var createDirective;
-        beforeEach(function () {
-            createDirective = function () {
-                helpers.createDirective(template, compile, scope);
-            };
-        });
-
-        describe('rx-form', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxForm.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 16]
```diff
@@ -1,417 +0,0 @@
...
-            before(function () {
-                template = '<form rx-form></form>';
-            });
-
-            it('should not throw error on creation', function () {
-                expect(createDirective).to.not.throw(Error);
-            });
-        });//rx-form
-
-        describe('rx-form-section', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxFormSection.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 17]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-form-section></rx-form-section>';
-                });
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-form-section
-
-        describe('rx-field', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxField.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 18]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-field></rx-field>';
-                });
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-field
-
-        describe('rx-field-name', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxFieldName.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 19]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-field-name></rx-field-name>';
-                });
-
-                it('should throw error when created', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-field-name
-
-        describe('rx-field-content', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxFieldContent.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 20]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-field-content></rx-field-content>';
-                });
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-field-content
-
-        describe('rx-input', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxInput.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 21]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-input></rx-input>';
-                });
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-input
-
-        describe('rx-prefix', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxPrefix.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 22]
```diff
@@ -1,417 +0,0 @@
...
-                    template = '<rx-prefix></rx-prefix>';
-                });
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-prefix
-
-        describe('rx-suffix', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxSuffix.spec.js`

File: src/rxForm/rxForm.spec.js [hunk 23]
```diff
@@ -1,417 +0,0 @@
...
-
-                it('should error', function () {
-                    expect(createDirective).to.throw(Error);
-                });
-            });
-        });//rx-suffix
-    });
-});
-
-describe('rxFormUtils', function () {
```
Comment (CITguy) [inline src/rxForm/rxForm.spec.js]: moved to `src/rxForm/scripts/rxFormUtils.spec.js`

**Comment by CITguy** [comment]
Added "DO NOT MERGE" because I missed a couple of files.

File: src/rxForm/scripts/rxForm.page.js [hunk 24]
```diff
@@ -158,6 +158,7 @@ exports.rxForm = {
         });
     },
 
+    //TODO: split out into exports.rxFieldName (src/rxForm/scripts/rxFieldName.page.js)
```
Comment (CITguy) [inline src/rxForm/scripts/rxForm.page.js]: [FRMW-422](https://jira.rax.io/browse/FRMW-442) created to capture todo.

**Comment by bry4n** [comment]
that was pretty big file for `rxForm` component. :smile: 

LGTM

**Comment by CITguy** [comment]
Yeah, rxForm and rxApp are the two largest monoliths.

**Comment by 100stacks** [comment]
@CITguy ditto what @bry4n said....LGTM :+1:

**Pull request merged**